src/include/platform/ThreadStackManager.h

```diff
@@ -106,6 +109,9 @@ class ThreadStackManager
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
 
+    CHIP_ERROR GetThreadNetworkDiagnosticAttributeInfo(chip::AttributeId attributeId, uint8_t ** buffer, uint16_t & ReadLength,
```

Comment:
Again, needs documentation.

And both here and in connectivity manager might be better for all the outparams to be pointers, not a mix of pointers and references....